Source/Core/DolphinQt/Config/ControllersWindow.cpp

```diff
@@ -426,7 +426,9 @@ void ControllersWindow::OnGCPadConfigure()
     return;
   }
 
-  MappingWindow(this, type, static_cast<int>(index)).exec();
+  MappingWindow* window = new MappingWindow(this, type, static_cast<int>(index));
```

Comment:
Does this leak the MappingWindow every time a controller mapping window is opened?

Reply:
I'm not sure.  I partially based it on the code in `MainWindow`, which doesn't ever explicitly delete it (though it only creates it once). From what I've looked up, [Qt will automatically free it](https://stackoverflow.com/a/2491811) when the parent is destroyed, but I'm not sure when that actually happens (`m_controllers_window` is also only created once in `MainWindow` and never explicitly freed).

I think at least that that means that it will be freed by the time dolphin is closed completely, possibly earlier... which might still not be great, as it would still leak for the whole lifetime of the application.  But I'm not sure it works that way.  I'll do some testing.

I could also set `WA_DeleteOnClose` for it, though that may cause issues for the one in MainWindow which is persistent (and I'm not sure whether that's safe here or not).

Reply:
Yeah then I think the windows you create won't be deleted until the entire app is closed. It should be safe to set the delete on close flag to fix that.

Reply:
Flag added; everything seems to work fine and it does resolve the leak (tested by adding a `char data[10000000];` into MappingWindow, and then just looking at memory levels; before it kept increasing for each one opened and after it increases when opened and then decreases after closed.  (The one in `MainWindow` is still not freed and remains present after it's opened once, but it's only allocated once.)